tests/src/Test/Tasty/Clash.hs

```diff
@@ -230,20 +199,20 @@ sequenceTests path (unzip -> (testNames, testTrees)) =
       applyAfter Nothing  tt = tt
       applyAfter (Just p) tt = after AllSucceed p tt
 
-data ClashTest = ClashTest
-  { ctExpectFailure :: Maybe (TestExitCode, T.Text)
+data ClashGen = ClashGen
```

Comment:
Not all of them are arguably tests though IMO. I can see arguments for `ClashGen` and `ClashExec` being tests, but `ClashBuild` is really just a step to produce the executable for `ClashExec` and not really a test in itself.

If we don't care about losing build and exec as different reported tests in the log, it might be possible to have just `ClashGenTest` and `ClashExecTest` and fold the `ClashBuild` and `ClashExec` types into one. That would also get rid of the nasty coupling of the definitions of `IsTest` for `ClashBuild` and `ClashExec`

Reply:
I agree not all of them are tests, but they're tests to tasty is what I meant.

> If we don't care about losing build and exec as different reported tests in the log

I like the ability to see all executed commands separately for reproducability. What do you think?

Reply:
Well the failure in `stderr` would very obviously tell you if it failed to build or run I think, so you don't really lose anything there IMO